### ISSUE TITLE
'kv.loader' cache is purged accidentally, causing key errors with app crash

### ISSUE THREAD

**Issue by DanTheMan2000** [kivy/kivy#6609]
### Versions

* Python: 3.7.3
* OS: Windows or Linux
* Kivy: 1.11.1

### Description

The entire 'kv.loader' cache is purged whenever a RecycleView tries to load an image that does not exist. In the below example, 'image1.png' does not exist, but is loaded in the `self.data` for the RecycleView. Since the image does not exist, the cache's key value is set to None. The application crashes whenever the 60 second cache timeout attempts to recycle the image. Whenever Cache.remove() is called, the key is None. In the code, a key value of None will purge the entire 'kv.loader' category. 

### Code and Logs

```python

from kivy.app import App
from kivy.lang import Builder
from kivy.properties import StringProperty
from kivy.uix.boxlayout import BoxLayout
from kivy.uix.recycleview import RecycleView
from kivy.uix.recycleview.views import RecycleDataViewBehavior

Builder.load_string('''

<ImageGalleryItem>:
    AsyncImage:
        size_hint: 1,1
        allow_stretch: True
        keep_ratio: True
        source: root.image_source
        
<ImageGallery>:
    id: rv
    viewclass: 'ImageGalleryItem'
    RecycleGridLayout:
        orientation: 'vertical'
        cols: 4
        spacing: '20dp'
        padding: [20]
        default_size: None, dp(rv.width/self.cols*.8)
        default_size_hint: 1, None
        size_hint_y: None
        height: self.minimum_height
''')


class ImageGalleryItem(RecycleDataViewBehavior, BoxLayout):
    image_source = StringProperty()


class ImageGallery(RecycleView):

    def on_kv_post(self, base_widget):
        # Create the dictionary of lists for RecycleView
        dict_list = []
        for image_path in self.get_image_paths():
            dict_list.append(dict({"image_source": image_path}))
        self.data = dict_list

    @staticmethod
    def get_image_paths():
        return ['image1.png', 'image2.png', 'image3.png']


class MainApp(App):

    def build(self):
        return ImageGallery()


if __name__ == '__main__':
    MainApp().run()


```
![image2](https://user-images.githubusercontent.com/52839433/69084484-f9ca7580-0a09-11ea-9ae6-581b15ee8db5.png)
![image3](https://user-images.githubusercontent.com/52839433/69084485-f9ca7580-0a09-11ea-9880-71002c5c0009.png)

**Comment by DanTheMan2000** [comment]
The red and blue images above are supposed to be image2.png and imag3.png. They need to be placed in the same directory as the script. In order for the crash to occur, some of the images must load after the image that does not exist.

**Comment by OutsourcedGuru** [comment]
Sounds a bit like [this](https://github.com/kivy/kivy/issues/6103#issuecomment-600859195), to be honest. Read the entire issue after you've read my work-around. In your case, you might do something like:

```
<RecycleView>:
    nocache: True
```

**Comment by tshirtman** [comment]
The merged PR #6815 might fix your issue, if you can test and report back. Thanks.
